x-pack/test/cases_api_integration/security_and_spaces/tests/common/comments/delete_comment.ts

```diff
@@ -91,9 +111,299 @@ export default ({ getService }: FtrProviderContext): void => {
+            auth,
+          });
+
+          await es.indices.refresh({ index: updatedAlerts.hits.hits.map((alert) => alert._index) });
+
+          const alertAfterDeletion = await getSecuritySolutionAlerts(
+            supertest,
+            updatedAlerts.hits.hits.map((alert) => alert._id)
+          );
+
+          const caseIdsWithoutRemovedCase =
+            expectedHttpCode === 204
+              ? updatedCases
+                  .filter((theCase) => theCase.id !== caseToDelete.id)
+                  .map((theCase) => theCase.id)
+              : updatedCases.map((theCase) => theCase.id);
+
+          for (const alert of alertAfterDeletion.hits.hits) {
+            expect(alert._source?.[ALERT_CASE_IDS]).eql(caseIdsWithoutRemovedCase);
+          }
+        };
+
+        beforeEach(async () => {
+          await esArchiver.load('x-pack/test/functional/es_archives/auditbeat/hosts');
+          await createSignalsIndex(supertest, log);
+        });
+
+        afterEach(async () => {
+          await deleteSignalsIndex(supertest, log);
+          await deleteAllRules(supertest, log);
+          await esArchiver.unload('x-pack/test/functional/es_archives/auditbeat/hosts');
+        });
+
+        it('removes a case from the alert schema when deleting an alert attachment', async () => {
+          await createCaseAttachAlertAndDeleteAlert({
+            totalCases: 1,
+            indexOfCaseToDelete: 0,
+          });
+        });
+
+        it('should remove only one case', async () => {
+          await createCaseAttachAlertAndDeleteAlert({
+            totalCases: 3,
+            indexOfCaseToDelete: 1,
+          });
+        });
+
+        it('should delete case ID from the alert schema when the user has write access to the indices and only read access to the siem solution', async () => {
+          await createCaseAttachAlertAndDeleteAlert({
+            totalCases: 1,
+            indexOfCaseToDelete: 0,
+            expectedHttpCode: 204,
+            auth: { user: secOnlyReadAlerts, space: 'space1' },
+          });
+        });
+
+        it('should NOT delete case ID from the alert schema when the user does NOT have access to the alert', async () => {
+          await createCaseAttachAlertAndDeleteAlert({
+            totalCases: 1,
+            indexOfCaseToDelete: 0,
+            expectedHttpCode: 403,
+            auth: { user: obsSec, space: 'space1' },
+          });
+        });
+
+        it('should delete the case ID from the alert schema when the user has read access to the kibana feature but no read access to the ES index', async () => {
+          await createCaseAttachAlertAndDeleteAlert({
+            totalCases: 1,
+            indexOfCaseToDelete: 0,
+            expectedHttpCode: 204,
+            auth: { user: secSolutionOnlyReadNoIndexAlerts, space: 'space1' },
+          });
+        });
+      });
+
+      describe('observability', () => {
+        const createCaseAttachAlertAndDeleteAlert = async ({
```

Comment:
Seems like this and the security solution function are pretty close do you think it'd make sense to move it to a function that both could call? I think the differences are the alerts that get created and owner, maybe those could be parameters?